sdk/keyvault/Azure.Security.KeyVault.Shared/src/ChallengeBasedAuthenticationPolicy.cs

```diff
@@ -75,16 +74,18 @@ private async ValueTask ProcessCoreAsync(HttpMessage message, ReadOnlyMemory<Htt
                 message.Request.Content = originalContent;
 
                 // update the cached challenge
-                var challenge = AuthenticationChallenge.GetChallenge(message);
+                challenge = AuthenticationChallenge.GetChallenge(message);
 
-                // if a challenge was returned and its different from the cached _challenge
-                if (challenge != null && !challenge.Equals(_challenge))
+                if (challenge != null)
```

Comment:
I think you still need the condition to check that the challenge is not the same, otherwise every 401 will result in a second attempt to send the request.